core/string/ustring.h

```diff
@@ -270,27 +270,13 @@ class String {
 	static constexpr char32_t _null = 0;
 	static constexpr char32_t _replacement_char = 0xfffd;
 
-	// Known-length copy.
-	void copy_from_unchecked(const char32_t *p_char, int p_length);
-
 	// NULL-terminated c string copy - automatically parse the string to find the length.
 	void append_latin1(const char *p_cstr) {
 		append_latin1(Span(p_cstr, p_cstr ? strlen(p_cstr) : 0));
 	}
 	void append_utf32(const char32_t *p_cstr) {
 		append_utf32(Span(p_cstr, p_cstr ? strlen(p_cstr) : 0));
 	}
```

Comment:
These should get `Error` return types as well, no? I'd also group these by their `Span` equivalents below (changing their return type as well) to match the other functions

Reply:
These are currently in the `private` section of `String`. I don't want to expose them, because I prefer callers having to use explicit `Span` syntax. 
I could still add an `Error` return type, though I don't think they're called from anywhere except the constructor at this point.